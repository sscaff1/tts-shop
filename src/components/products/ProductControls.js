import React from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { Button } from 'antd';

import cartActions from 'actions/cart';

function ProductControls({ id }) {
  const quantity = useSelector(state => state.cart[id] || 0);
  const dispatch = useDispatch();
  if (quantity === 0) {
    return (
      <Button onClick={() => dispatch(cartActions.add(id))}>Add to Cart</Button>
    );
  }
  return (
    <div>
      <Button onClick={() => dispatch(cartActions.subtract(id))}>-</Button>
      <span style={{ marginLeft: 5, marginRight: 5 }}>{quantity}</span>
      <Button onClick={() => dispatch(cartActions.add(id))}>+</Button>
    </div>
  );
}

ProductControls.propTypes = {
  id: PropTypes.string.isRequired,
};

export default ProductControls;
