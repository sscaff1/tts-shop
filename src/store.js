import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducer';

const initState = {
  products: [],
  ui: {
    loading: false,
  },
  cart: {},
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default (initialState = initState) =>
  createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk))
  );
