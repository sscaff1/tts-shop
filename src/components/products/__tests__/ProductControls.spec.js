import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import ProductControls from '../ProductControls';
import store from '../../../store';

describe('Product Controls', () => {
  const setup = () => {
    return (
      <Provider store={store}>
        <ProductControls id="1" />
      </Provider>
    );
  };
  beforeEach(cleanup);

  it('should render', () => {
    const { container } = render(setup());
    expect(container).toMatchSnapshot();
  });

  it('should render a count and two action buttons when initial is pressed', () => {
    const { container } = render(setup());
    const initialButton = container.querySelector('button');

    fireEvent.click(initialButton);

    expect(container).toMatchSnapshot();
  });
});
