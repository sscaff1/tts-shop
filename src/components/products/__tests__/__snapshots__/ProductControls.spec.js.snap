// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Controls should render 1`] = `
<div>
  <button
    class="ant-btn"
    type="button"
  >
    <span>
      Add to Cart
    </span>
  </button>
</div>
`;

exports[`Product Controls should render a count and two action buttons when initial is pressed 1`] = `
<div>
  <div>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        -
      </span>
    </button>
    <span
      style="margin-left: 5px; margin-right: 5px;"
    >
      1
    </span>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        +
      </span>
    </button>
  </div>
</div>
`;
